New UI for receive payment flows

erdemyerebasmaz opened the pull request:
This PR unifies receive payment flows under the same page.

### Changelist:
- Unify receive payment methods under same UI d8194ec
  - All receive payment pages are moved under `receive_payment` dir and `ReceivePaymentPage`, which takes a `pageIndex` and displays the selected payment methods page. Renamed files to reflect these changes.
  - UI's are made consistent across all payment pages where possible. 
    - All pages now use the same `AddressWidget` to display payment information.
    - They use similar fee information widgets etc.
  - Dialogs on payment flows are removed or integrated to a fulll page UI as part of their respective flow.
- Open the bottom sheet menu item via ReceivePaymentPage 3e586b7
- Remove payment method selector 69a9505
#### Layout & Theme issues:
- Do not override ExpiryAndFeeMessage colors on light theme  edcc375
- Move padding to ReceivePayment page 241068a
- Fix share & copy icon colors 077ded2
- Fix loader alignment on receive pages 8dc0e40
- Fix header displacement after QR code loads fc40ccb

#### Misc:
- Added Breez icon to QR image
- Replace hardcoded messages from Breez-Translations d7e6b25
- Add helper methods for readability d46d4da
  - These will be extracted into their own widgets
  
  
 ### TODO:
 - [x] Track payments and close the payment information to display bubble animation on home page when the payment is received 35db836

dangeross commented on lib/routes/receive_payment/lightning/widgets/compact_qr_image.dart:
```diff
@@ -49,7 +49,7 @@ class CompactQRImage extends StatelessWidget {
   final bool bip21;
 
-  const CompactQRImage({super.key, required this.data, this.size, this.bip21 = false});
+  const CompactQRImage({super.key, required this.data, this.size, this.bip21 = true});
```
Assuming this is a bolt11 invoice (lightning widget), should bip21 be set to `false`?

dangeross commented on lib/routes/receive_payment/onchain/bitcoin_address/receive_bitcoin_address_payment_page.dart:
```diff
@@ -0,0 +1,239 @@
+        return AddressWidget(
+          snapshot: snapshot,
+          title: texts.withdraw_funds_btc_address,
+          type: AddressWidgetType.lightning,
```
Just checking, should this be `AddressWidgetType.bitcoin`?

erdemyerebasmaz replied:
Addressed as part of: [Unify fee & limits message box widgets](https://github.com/breez/misty-breez/pull/167/commits/75bccf2f0c3639fec6a44826c0dd80367db61edd)

dangeross commented on lib/routes/receive_payment/onchain/liquid_address/receive_liquid_address_payment_page.dart:
```diff
@@ -0,0 +1,315 @@
+import 'package:share_plus/share_plus.dart';
+
+class ReceiveLiquidAddressPaymentPage extends StatefulWidget {
+  static const routeName = "/receive_bitcoin_address";
```
```suggestion
  static const routeName = "/receive_liquid_address";
```

erdemyerebasmaz replied:
Removed `ReceiveLiquidAddressPaymentPage` as it's unused 56df30ea90ce63a171c86d312807e7b23d96db4f

dangeross commented on lib/routes/receive_payment/onchain/liquid_address/receive_liquid_address_payment_page.dart:
```diff
@@ -0,0 +1,315 @@
+              ),
+            );
+          }
+          if (snapshot.onchainPaymentLimits == null) {
```
I know we're not using it, but we don't need onchainPaymentLimits for Liquid

dangeross commented on lib/routes/receive_payment/widgets/address_widget/address_qr_widget.dart:
```diff
@@ -0,0 +1,56 @@
+          ? const SizedBox.shrink()
+          : Column(
+              children: [
+                AddressQR(bolt11: address ?? snapshot!.data!.destination, bip21: true),
```
Should the `bip21` value depend on the AddressWidgetType?

erdemyerebasmaz replied:
Removed uppercase QR trick to produce smaller QR codes.

dangeross commented on lib/routes/receive_payment/widgets/address_widget/address_qr.dart:
```diff
@@ -0,0 +1,24 @@
+import 'package:l_breez/routes/receive_payment/lightning/widgets/compact_qr_image.dart';
+
+class AddressQR extends StatelessWidget {
+  final String bolt11;
```
Maybe rename this to `destination` or `data`

erdemyerebasmaz replied:
I agree, these widgets will be renamed & refactored next.

erdemyerebasmaz replied:
Done. [Refactor AddressWidget](https://github.com/breez/misty-breez/pull/167/commits/ea1ef395a50d2e950f0615a8653dbfb07a807abd)

erdemyerebasmaz commented:
@dangeross New changes since last review:

#### Payment tracking & successful payment animation
- Track payments and pop to home page to display bubble animation on receiving payment 
- Only track incoming payments
- Fade out the bubble animation and pop the SuccessfulPaymentRoute 
- Display payment received message upon receiving a payment
- Added comments for flip transition animation
- Removed `didUpdateWidget` to avoid displaying `FlipTransition` twice

#### DestinationWidget changes
- Refactored `AddressWidget`
- Add padding to payment limits message box loader
- Renamed and moved address widget folders

#### LNURL payment
- Display errors for LNURL payments that fall below payment limits
- Fix the minimum value pasted on clicking limits label on LNURL payment page
- Use domain from `LnUrlPayRequestData` on it's title

#### LNURL Withdraw
- Support LNURL Withdraw with new UI 
- Created a dedicated LnUrl Withdraw page
- Display errors for URLs recognized through QR scanner
- Fix amount auto focus on `LnUrlWithdrawPage`
- Remove LnUrlWithdraw logic from `ReceiveLightningPaymentPage`

dangeross commented on lib/routes/receive_payment/lnurl/lnurl_withdraw_page.dart:
```diff
@@ -0,0 +1,344 @@
+        final minWithdrawableFormatted = currencyState.bitcoinCurrency.format(minWithdrawable);
+        final maxWithdrawableFormatted = currencyState.bitcoinCurrency.format(maxWithdrawable);
+
+        final isFixedAmountWithinLimits = minWithdrawable == maxWithdrawable;
+        if (!isFixedAmountWithinLimits) {
```
Is this right? Seems inconsistent with the error text and the min/max is already checked

erdemyerebasmaz replied:
Explained in Slack and [added a different message if fixed payment amount is outside allowed limits](https://github.com/breez/misty-breez/pull/167/commits/ab87441912a0365c837c9e250447fd3db6ad5b35)

dangeross commented on lib/routes/receive_payment/lnurl/lnurl_withdraw_page.dart:
```diff
@@ -44,36 +42,75 @@ class LnUrlWithdrawPageState extends State<LnUrlWithdrawPage> {
+      final currencyCubit = context.read<CurrencyCubit>();
+      final currencyState = currencyCubit.state;
+
+      final isFixedAmountWithinLimits = _isFixedAmount && (effectiveMinSat == effectiveMaxSat);
```
I think `_isFixedAmount` has not been set yet as `_initializePageData()` is called after
